app/javascript/components/Login/validation.js

```diff
@@ -0,0 +1,13 @@
+export const LoginValidation = (form) => {
+  let error = {};
+
+  const { email, password} = form;
+
+  if(!email) error.email = "Invalid Email!";
+  if(!password) error.password = "Invalid password!";
+
+  return {
+    valid: Object.keys(error).length === 0,
```

Comment:
What does this object refer to?

Reply:
This object refers to the errors of each field and we could store the customed message. `valid` for checking if there is no error in any fields then it should be `true` such that we don't need to check again.